server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -874,6 +877,7 @@ public void flush(long tableFlushID) {
         } else {
           initiateMinor = true;
         }
+        updateCounter.getAndIncrement();
```

Comment:
I think it would be better to place this counter in the DataFileManager class.  Also would not make it an atomic long, would just make it a long.  The variable it tightly coupled to the map dataFileSizes in DataFileManager.  Any time that map is update, the counter needs to be incremented.  This map update and counter increment should all be done in the same sync block, so there is no need to make the counter an atomic long.  Making it an atomic long gives the impression that it can be accessed on its own when in reality it should only be accessed in the same sync that controls access to the map.

The DataFileManager class relies on being accessed when the tablet lock is held, so it does not really need its own synch.

So could add the counter to the DataFileManager class and updated whenever the map is updated and also add a method to get the counter from DataFileManager.  Rely on the tablet lock for sync, there is another comment related to this.

Reply:
Addressed in 4f282f4